linera-core/src/client/mod.rs

```diff
@@ -2108,12 +2123,20 @@ where
     #[instrument(level = "trace", skip(incoming_bundles, operations))]
     async fn set_pending_block(
```

Comment:
Maybe we can rename this method to `new_pending_block`? Or maybe you have even better idea.

The reason I'm suggesting is:
- It's not _setting_ per-se, it's creating and then setting. But it can still fail so it may not even set anything.
- `set_pending_block` calls another `set_pending_block` method and I don't really like having many methods with exact same name